newIDE/app/src/GameDashboard/GameRegistration.js

```diff
@@ -116,22 +121,102 @@ export const GameRegistration = ({
     [getAuthorizationHeader, profile, project, loadGame, onGameRegistered]
   );
 
+  const _onAcceptGameStatsEmail = React.useCallback(
+    async () => {
+      if (!profile || !project) return;
+
+      setAcceptGameStatsEmailInProgress(true);
+      try {
+        await onAcceptGameStatsEmail();
+      } catch (error) {
+        console.error('Unable to acccept game stats email.', error);
+        showErrorBox({
+          rawError: error,
+          errorId: 'game-stats-email-error',
+          message:
+            'Unable to accept game stats email. ' +
+            ' ' +
+            'Verify your internet connection or try again later.',
+        });
+      }
+      setAcceptGameStatsEmailInProgress(false);
+    },
+    [profile, project, onAcceptGameStatsEmail]
+  );
+
   React.useEffect(
     () => {
       loadGame();
     },
     [loadGame]
   );
 
-  if (!authenticated) {
+  if (!project || hideIfRegistered) {
```

Comment:
Regression here it seems: If `hideIfRegistered === true`, this component will be never rendered.